server/bin/pbench-base.py

```diff
@@ -28,18 +29,16 @@
 )
 parsed, _ = parser.parse_known_args()
 
-_prog = os.path.basename(parsed.prog[0])
-_dir = os.path.dirname(parsed.prog[0])
+_prog = Path(parsed.prog[0]).name
+_dir = Path(parsed.prog[0]).parent
 
 if not parsed.cfg_name:
     # pbench-base.py is not always invoked with -C or --config or the _PBENCH_SERVER_CONFIG
     # environment variable set.  Since we really need access to the config
     # file to operate, and we know the relative location of that config file,
     # we check to see if that exists before declaring a problem.
-    config_name = os.path.join(
-        os.path.dirname(_dir), "lib", "config", "pbench-server.cfg"
-    )
-    if not os.path.exists(config_name):
+    config_name = Path(_dir.parent, "lib", "config", "pbench-server.cfg")
```

Comment:
I would feel better if `config_name` was consistenly either a string or a path: it's a path in this branch, it's a string in the else branch. I tested that `PbenchConfig()` is happy with either a string or a path, but we should be consistent (and if you decide to make it a path in all cases, then rename it to reflect that choice).

Reply:
ok sure